php/composer.json

```diff
@@ -0,0 +1,33 @@
+{
+    "name": "truelayer/signing",
+    "description": "Produce and verify TrueLayer API requests signatures",
+    "type": "library",
+    "license": "MIT",
+    "autoload": {
+        "psr-4": {
+            "TrueLayer\\Signing\\": "src/"
+        }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "TrueLayer\\Signing\\Tests\\": "tests"
+        }
+    },
+    "authors": [
+        {
+            "name": "Stefan Adrian Danaita",
+            "email": "me@dsa.io"
+        }
+    ],
+    "require": {
+        "php": "^7.4|^8.0",
+        "web-token/jwt-core": "^2.2",
+        "web-token/jwt-signature": "^2.2",
+        "web-token/jwt-signature-algorithm-ecdsa": "^2.2",
+        "web-token/jwt-key-mgmt": "^2.2"
+    },
+    "require-dev": {
+        "pestphp/pest": "^1.20",
```

Comment:
I would recommend adding `phpstan` or `psalm` here, it's a static analysis tool for finding bugs in the code. See https://phpstan.org/user-guide/getting-started

I've just run phpstan at the strictest settings and it's found a few issues:

```
vendor/bin/phpstan analyse --level 9 src tests
 15/15 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ----------------------------------------------------------------------------------------------------------------------------- 
  Line   src/AbstractJws.php                                                                                                          
 ------ ----------------------------------------------------------------------------------------------------------------------------- 
  13     Property TrueLayer\Signing\AbstractJws::$request_headers type has no value type specified in iterable type array.            
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                     
  39     Method TrueLayer\Signing\AbstractJws::headers() has parameter $headers with no value type specified in iterable type array.  
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                     
 ------ ----------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------- 
  Line   src/Contracts/Jws.php                                                                                                          
 ------ ------------------------------------------------------------------------------------------------------------------------------- 
  11     Method TrueLayer\Signing\Contracts\Jws::headers() has parameter $headers with no value type specified in iterable type array.  
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                       
 ------ ------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   src/Contracts/Verifier.php                                                                                                                 
 ------ ------------------------------------------------------------------------------------------------------------------------------------------- 
  14     Method TrueLayer\Signing\Contracts\Verifier::requireHeaders() has parameter $headers with no value type specified in iterable type array.  
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                                   
 ------ ------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   src/Verifier.php                                                                                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  29     Property TrueLayer\Signing\Verifier::$required_headers type has no value type specified in iterable type array.                                                                                     
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                                                                                            
  66     Method TrueLayer\Signing\Verifier::requireHeaders() has parameter $headers with no value type specified in iterable type array.                                                                     
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                                                                                            
  81     PHPDoc tag @throws with type                                                                                                                                                                        
         TrueLayer\Signing\Exceptions\InvalidAlgorithmException|TrueLayer\Signing\Exceptions\InvalidSignatureException|TrueLayer\Signing\Exceptions\InvalidTrueLayerSignatureVersionException|TrueLayer\Sig  
         ning\Exceptions\RequestPathNotFoundException|TrueLayer\Signing\Exceptions\RequiredHeaderMissingException|TrueLayer\Signing\Exceptions\TrueLayerHeadersNotFoundException is not subtype of           
         Throwable                                                                                                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------- 
  Line   tests/MockData.php                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------- 
  8      Method TrueLayer\Signing\Tests\MockData::generateKeyPair() return type has no value type specified in iterable type array.  
         💡 See: https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------- 
  Line   tests/SignerTest.php                                                                          
 ------ ---------------------------------------------------------------------------------------------- 
  9      Static method TrueLayer\Signing\Signer::signWithKey() invoked with 3 parameters, 2 required.  
  12     Access to an undefined property Pest\Expectation|Pest\Support\Extendable::$not.               
  13     Call to an undefined method Pest\Expectation|Pest\Support\Extendable::toBeString().           
  18     Static method TrueLayer\Signing\Signer::signWithKey() invoked with 3 parameters, 2 required.  
 ------ ---------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------- 
  Line   tests/VerifierTest.php                                                                        
 ------ ---------------------------------------------------------------------------------------------- 
  10     Static method TrueLayer\Signing\Signer::signWithKey() invoked with 3 parameters, 2 required.  
  13     Call to an undefined method TrueLayer\Signing\AbstractJws::sign().                            
  19     Call to an undefined method TrueLayer\Signing\AbstractJws::requireHeaders().                  
  27     Access to an undefined property Pest\Expectation|Pest\Support\Extendable::$not.               
  27     Result of method TrueLayer\Signing\Verifier::verify() (void) is used.                         
  32     Static method TrueLayer\Signing\Signer::signWithKey() invoked with 3 parameters, 2 required.  
  35     Call to an undefined method TrueLayer\Signing\AbstractJws::sign().                            
  41     Call to an undefined method TrueLayer\Signing\AbstractJws::requireHeaders().                  
  57     Static method TrueLayer\Signing\Signer::signWithKey() invoked with 3 parameters, 2 required.  
  60     Call to an undefined method TrueLayer\Signing\AbstractJws::sign().                            
  66     Call to an undefined method TrueLayer\Signing\AbstractJws::requireHeaders().                  
 ------ ---------------------------------------------------------------------------------------------- 

                                                                                                                        
 [ERROR] Found 23 errors                                                                                                
```

Reply:
Could also consider adding https://github.com/Roave/SecurityAdvisories as well